### ISSUE TITLE
Have mksurfdata_esmf pick up the mpi command from env_mach_specific.xml

### ISSUE THREAD

**Issue by slevisconsulting** [ESCOMP/CTSM#1776]
Motivated by this [comment](https://github.com/ESCOMP/CTSM/pull/1756#issuecomment-1138146016):
The best way of keeping the mpi command up-to-date and working on various machine/compiler/mpilib combinations is to query the env_mach_specific.xml file for the mpirun command. The mpirun command is fairly complex and depends on the MPI library being used.

E.g. for mvapich2
`mpiexec --machinefile $ENV{PBS_NODEFILE} -n <ntasks> --prepend-rank`
and for openmpi
`mpiexec -n <ntasks> --tag-output`

@ekluzek I thought I hadn't resolved my issues with PBS_NODEFILE, but this [comment](https://github.com/ESCOMP/CTSM/pull/1748#issuecomment-1139101946) and the one after it suggest that I got past it.

**Comment by slevisconsulting** [comment]
@jedwards4b I will start by asking you and let me know if I should be asking somebody else.

I'm reading the `env_mach_specific.xml` file to piece together mpirun commands for the new mksurfdata_esmf tool on different machines. In the next lines I show what I the .xml file provides versus what I need for my python code to work:

On cheyenne I get the first but want the second:
```
mpiexec_mpt -p "%g:" -np {{ total_tasks }}
mpiexec_mpt -p \"%g:\" -np {total_tasks}
```

On casper:
```
mpirun  -np {{ total_tasks }}
mpirun  -np {total_tasks}
```

On izumi:
```
mpiexec --machinefile $ENV{$PBS_NODEFILE} -n {{ total_tasks }}
mpiexec --machinefile $PBS_NODEFILE -n {total_tasks}
```

The question is about changing the strings in the .xml to look the way I need them. Would that break things elsewhere? If not, then that seems like the ideal solution. If things elsewhere will break, then I will need to manipulate the strings in my code.

**Comment by jedwards4b** [comment]
At a higher level there is a case object [get_mpirun_cmd ](https://github.com/ESMCI/cime/blob/master/CIME/case/case.py#L1960).   In either you use the overrides argument to set the values for stuff in {{ }}
For example overrides = {"total_tasks" : 8,} will replace {{ total_tasks }} with the value 8.

**Comment by slevisconsulting** [comment]
That's probably what I want, thank you! (I was putting together my own code to read the .xml, not using cime.)

**Comment by jedwards4b** [comment]
Regarding putting together your own code - I think that is dangerous.  The xml files in cime are subject to change in a non-backward compatible way, we promise that interfaces that use cime to read them will always work, if you read it yourself (outside of cime) you may cause yourself some maintenance headaches.

**Comment by ekluzek** [comment]
@jedwards4b the thing about this work is that we currently our outside of a cime case. We've built the build using the cime configure to create env_mach_specific.xml and some cmake files. But, it doesn't create a case.

We've been nervous about using cime code because we likely will have to create a case, which is a bit of overkill for this use case. I do see your point about the risk of doing our own code, but I'm not sure we can just read the env_mach_specific.xml file from cime without doing all the work to create a case, which will then have a bunch of meaningless files for this usage. So it also might obfuscate what's going on for the build here.

**Comment by jedwards4b** [comment]
I think that you can do this without a case and there are examples of doing so in cime.

**Comment by jedwards4b** [comment]
Since you have a env_mach_specific.xml file you should be able to instantiate an object of type EnvMachSpecific
using the directory containing that file as caseroot.   See CIME/BuildTools/configure.py for an example.

**Comment by slevisconsulting** [comment]
My `/glade/work/slevis/git/vert_res_in_mimics/cime/CIME/BuildTools` 
and `/glade/work/slevis/git/mksurfdata_toolchain/cime/CIME/BuildTools`
both only contain `__pycache__`
Am I looking in the wrong place for the example @jedwards4b ?

I have found some [documentation about CIME.BuildTools here](http://esmci.github.io/cime/versions/master/html/CIME_api/CIME.BuildTools.html#), but without the example I will need some help to understand what I need to do...

**Comment by slevisconsulting** [comment]
I think I'm closer. Looking here now:
http://esmci.github.io/cime/versions/master/html/_modules/CIME/BuildTools/configure.html#FakeCase

**Comment by slevisconsulting** [comment]
Thank you @jedwards4b 
I got it to work: https://github.com/ESCOMP/CTSM/pull/1796/commits/fb87525e7c6e1ae8092afbe6d643239e22f000a9

**Comment by slevisconsulting** [comment]
Closed with merge of #1796
